test/text/vtt_text_renderer_unit.js

```diff
@@ -0,0 +1,103 @@
+/** @license
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+describe('VTT tag rendering', () => {
+  /**
+   * Generates a rendered HTMLElement from a single cue text.
+   *
+   * @param {string} cueText
+   * @return {!Promise<HTMLElement>}
+   * @private
+   */
+  const renderCue = async (cueText) => {
+    const container = /** @type {!HTMLElement} */(
+      document.createElement('div'));
+    const video = new shaka.test.FakeVideo(0);
+    const displayer = new shaka.text.UITextDisplayer(video, container);
+    displayer.setTextVisibility(true);
+
+    displayer.append([new shaka.text.Cue(0, 100, cueText)]);
+    await shaka.test.Util.delay(0.5);
```

Comment:
This is a long delay for a unit test.  How about just calling the private function so they can be run synchronously?  You can use `/** @suppress {visibility} */` on a function to allow calling private functions.